src/cls/IPM/Storage/Module.cls

```diff
@@ -85,43 +85,54 @@ Property AfterInstallMessage As %String(MAXLEN = "", XMLPROJECTION = "Element");
 
 Property Deployed As %Boolean(XMLPROJECTION = "Element");
 
-// Relationship UpdateSteps As %IPM.Storage.UpdateStep(XMLPROJECTION = "NONE") [ Cardinality = one, Inverse = manyProp, OnDelete = cascade ];
+Relationship UpdateSteps As %IPM.Storage.UpdateStep(XMLPROJECTION = "NONE") [ Cardinality = many, Inverse = ModuleHome ];
 
 Property UpdatePackage As %String(MAXLEN = 255);
 
-ClassMethod AddUpdateStep(
+/// Add an update step and connect it to this module
+Method AddUpdateStep(
 	className As %String,
-	methodName As %String) As %Status
+	methodName As %String) As %IPM.Storage.UpdateStep
 {
-	return ##class(%IPM.Storage.UpdateStep).%New(className, methodName)
+	set newStep = ##class(%IPM.Storage.UpdateStep).%New(className, methodName)
+	do ..UpdateSteps.Insert(newStep)
+	do ..%Save()
+	return newStep
 }
 
+/// Seed update steps for all versions prior to the current major version by setting TimeStampEnd to current time
+/// Steps that already have TimeStampEnd defined (already been run) will be skipped
 Method SeedUpdateSteps() As %Status
 {
-	set majorVersion = ..Version.Major
 	set uniqueInstanceName = ##class(%SYS.System).GetUniqueInstanceName()
-    set mirroredHostName = $piece(uniqueInstanceName, ":", 1)
-    set mirroredInstanceName = $piece(uniqueInstanceName, ":", 2)
+	set mirroredHostName = $piece(uniqueInstanceName, ":", 1)
```

Comment:
Nit: no need to call these mirrored since these have nothing to do with mirroring. Also, you can use GetNodeName() and GetInstanceName() instead of GetUniqueInstanceName() and using $piece which is less clear IMO